### ISSUE TITLE
request.ssl() returning None from Server::https() requests

### ISSUE THREAD

**Issue by mikedilger** [hyperium/hyper#678]
This sort of thing documented on the Request page:
```
extern crate openssl;
use openssl::ssl::SslStream;
use hyper::net::HttpStream;
let maybe_ssl = req.ssl::<SslStream<HttpStream>>();
```
always yields None when I try it, even though I'm sure the request came via SSL from a server started with Server::https() and a browser that gives me a nice green ssl lock icon.

I was hoping to use this functionality to get the CN from client side certificates as a method of authentication.

**Comment by MarkSwanson** [comment]
This may help:

https://github.com/hyperium/hyper/issues/725

**Comment by seanmonstar** [comment]
I've fixed this in the 0.9.x branch, it will be part of the next bug fix release.
